### PR TITLE
create validate-release-note-json gh action

### PR DESCRIPTION
**Pull request by tabathadelane**
- runs on PRs to develop
- verifies that every agent has release notes
- verifies our expected fields exist

Related: [synthetics monitor](https://staging.onenr.io/0kLwGb4bAQ6) that checks the status of the docs link associated with each agents most recent release note to help us foresee potential JSON/Gatsby builds being out of sync.

### CODE REVIEW

File: scripts/releaseNotes.mjs [hunk 1]
```diff
@@ -105,6 +107,52 @@ const releaseNotes = (
...
+  json.forEach((note) =>
+    requiredData.forEach((key) => {
+      if (!note[key]) {
+        console.error(note);
+        console.error(`😵 Missing ${key} data`);
+        process.exitCode(1);
+      }
+    })
+  );
+  console.error(`✨ Release notes JSON validated`);
```
Comment (sunnyzanchi) [inline scripts/releaseNotes.mjs:153]: i think we should put a conditional here with a different message if validation fails. rn i think this kinda of reads like the notes are valid whether they pass or not

File: scripts/releaseNotes.mjs [hunk 2]
```diff
@@ -105,6 +107,52 @@ const releaseNotes = (
 );
 console.error('📦 release notes JSON generated');
 
+const validateReleaseNotesAgents = (json) => {
```
Comment (sunnyzanchi) [inline scripts/releaseNotes.mjs:110]: maybe a nit but i think we should name this arg `releaseNotes`. `json` might imply that it's already stringified

```suggestion
const validateReleaseNotesAgents = (releaseNotes) => {
```

**Review by sunnyzanchi — CHANGES_REQUESTED** [review]

File: scripts/releaseNotes.mjs (hunk 1 above)
Comment (tabathadelane) [reply scripts/releaseNotes.mjs:153]: Do you mean for eng readability? or for the Actions console? This line of code actually doesn't run if it [fails](https://github.com/newrelic/docs-website/actions/runs/6228948584/job/16906611920).

Comment (tabathadelane) [reply scripts/releaseNotes.mjs]: updated! I actually like [this](https://github.com/newrelic/docs-website/actions/runs/6241138043/job/16942723937) better

**Review by sunnyzanchi — APPROVED** [review]

**Pull request merged**